### ISSUE TITLE
continue refactoring: cpu enable_feature_grouping parameter is possibly useless

### ISSUE THREAD

**Issue by ShvetsKS** [dmlc/xgboost#7016]
Hi, after running several benchmarks I noticed that there are no any benefits (in terms of accuracy and/or especially performance) from usage of `enable_feature_grouping` parameter in cpu branch:

training time:
| dataset| enable_feature_grouping: 0 | enable_feature_grouping: 1 |
| ------------- |-------------| -----|
| airline-ohe  (1m rows) | 36.37s | 860.75s |
| higgs (1m rows) | 15.46s | 108.13s |
| mortgage (first Q)  | 40.54s | 237.94s |
| msrank  | 72.68s | 892.15s |
| santander | 135.67s | 417.11s |

**accuracy is the same** for both cases (enable_feature_grouping: 0, 1)

There is **~10x** performance slowdown on average

Are there any cases where `enable_feature_grouping` parameter gives any benefits (or it works as not expected)?

If no, in the context of `cpu_hist` refactoring process I'd suggest to remove all related to `enable_feature_grouping` parameter code (ex.: `BuildBlockHist`, `GHistIndexBlockMatrix` etc.).

Another reason of raising this issue/discussion is absence of `enable_feature_grouping` parameter description in official documentation.

**Comment by trivialfis** [comment]
It's supposed to handle sparse dataset created by one hot encoding I think?  @hcho3

**Comment by hcho3** [comment]
@trivialfis I crreated it to mimic the Exclusive Feature Bundling (EFB) feature in LightGBM, as described in https://papers.nips.cc/paper/2017/file/6449f44a102fde848669bdd9eb6b76fa-Paper.pdf. It's an experimental feature that really didn't work out.

**Comment by ShvetsKS** [comment]
> parse dataset created by one hot encoding I think? @hch

I tested on airline one hot encoded and it didn't work well (first line of the table)

**Comment by hcho3** [comment]
I suggest we remove `enable_feature_grouping`.

**Comment by ShvetsKS** [comment]
@hcho3 Thanks for fast decision!
I can prepare PR in a few hours
